File: 1352_product_of_the_last_k_numbers.java
import java.util.*;

class ProductOfNumbers {
    private List<Integer> list;

    public ProductOfNumbers() {
        list = new ArrayList<>();
        list.add(1);
    }

    public void add(int num) {
        if (num == 0) {
            list.clear();
            list.add(1);
        } else {
            list.add(list.get(list.size() - 1) * num);
        }
    }

    public int getProduct(int k) {
        int n = list.size();
        return n > k ? list.get(n - 1) / list.get(n - k - 1) : 0;
    }
}

File: 1352_product_of_the_last_k_numbers.py
class ProductOfNumbers:
    def __init__(self):
        self.lst = [1]

    def add(self, num):
        if num == 0:
            self.lst = [1]
        else:
            self.lst.append(self.lst[-1] * num)

    def getProduct(self, k):
        return 0 if len(self.lst) <= k else self.lst[-1] // self.lst[-k-1]
